pytorch_to_returnn/torch/nn/functional.py

```diff
@@ -799,6 +807,12 @@ def multi_head_attention_forward(
     return attn_output, None
 
 
+def cosine_similarity(x1: Tensor, x2: Tensor, dim: Optional[int] = 1, eps: float=1e-8) -> Tensor:
+  numerator = modules.Dot()(x1, x2, reduce_dim_a=dim, reduce_dim_b=dim)
+  inv_denominator = min(cat([rsqrt(pow(x1, 2).sum() * pow(x2, 2).sum()), tensor(eps ** -1)]))
```

Comment:
Use `minimum` instead.

Reply:
To which RETURNN layer would we map that?

Reply:
Maybe simply via `EvalLayer` and `sources: [a, b], eval: "tf.minimum(source(0), source(1))"`